datafusion/src/datasource/file_format/json.rs

```diff
@@ -38,20 +40,49 @@ use crate::physical_plan::Statistics;
 
 /// New line delimited JSON `FileFormat` implementation.
 pub struct JsonFormat {
-    /// If no schema was provided for the table, it will be
-    /// infered from the data itself, this limits the number
-    /// of lines used in the process.
-    pub schema_infer_max_rec: Option<usize>,
+    schema_infer_max_rec: Option<usize>,
+    object_store_registry: Arc<ObjectStoreRegistry>,
+}
+
+impl Default for JsonFormat {
+    fn default() -> Self {
+        Self {
+            schema_infer_max_rec: None,
+            object_store_registry: Arc::new(ObjectStoreRegistry::new()),
+        }
+    }
+}
+
+impl JsonFormat {
+    /// Create Parquet with the given object store and default values
+    pub fn new(object_store_registry: Arc<ObjectStoreRegistry>) -> Self {
+        Self {
+            object_store_registry,
+            ..Default::default()
+        }
+    }
+
+    /// Set a limit in terms of records to scan to infer the schema
+    /// - defaults to `None` (no limit)
+    pub fn with_schema_infer_max_rec(&mut self, max_rec: Option<usize>) -> &mut Self {
+        self.schema_infer_max_rec = max_rec;
+        self
+    }
 }
 
 #[async_trait]
 impl FileFormat for JsonFormat {
-    async fn infer_schema(&self, mut paths: StringStream) -> Result<SchemaRef> {
+    async fn infer_schema(&self, mut file_stream: SizedFileStream) -> Result<SchemaRef> {
         let mut schemas = Vec::new();
         let mut records_to_read = self.schema_infer_max_rec.unwrap_or(usize::MAX);
-        while let Some(file) = paths.next().await {
-            let file = File::open(file)?;
-            let mut reader = BufReader::new(file);
+        while let Some(fmeta_res) = file_stream.next().await {
+            let fmeta = fmeta_res?;
+            let reader = self
+                .object_store_registry
+                .get_by_uri(&fmeta.path)?
```

Comment:
given that `SizedFileStream` is returned from object store list call, I believe their paths would not contain uri scheme, so `get_by_uri` will always return the local file store? The naming convention that could be useful to establish within the code base would be using uri for string that could contain scheme while path for relative path component from the uri, i.e. without scheme.

Reply:
Very good point. It wasn't clear for me that the returned values from list calls wouldn't be URIs. I'll try to make that clearer.